### ISSUE TITLE
Variadic overrides reported to be incompatible

### ISSUE THREAD

**Issue by layday** [microsoft/pyright#1391]
**Describe the bug**
An overridden method with `*args` and `**kwargs` is reported to be incompatible.  Overrides with star arguments should be considered to be safe where the variadic type is a supertype of all of the parent method's parameters which are not declared in the override.

**To Reproduce**
Type check the snippet below.

**Expected behavior**
That no error would be produced.

**Screenshots or Code**
```python
# pyright: strict

class Foo:
    def bar(self, a: str, b: int, c: float, d: bool) -> None:
        ...

class Baz(Foo):
    # Method "bar" overrides class "Foo" in an incompatible manner
    #   Parameter count mismatch; base method has 5, but override has 3
    #   Parameter 2 name mismatch: base parameter is named "a", override parameter is named "args"
    #   Parameter 3 name mismatch: base parameter is named "b", override parameter is named "kwargs"
    def bar(self, *args: object, **kwargs: object) -> None:
        ...

class Qux(Foo):
    # Method "bar" overrides class "Foo" in an incompatible manner
    #   Parameter count mismatch; base method has 5, but override has 4
    #   Parameter 3 name mismatch: base parameter is named "b", override parameter is named "args"
    #   Parameter 4 name mismatch: base parameter is named "c", override parameter is named "kwargs"
    def bar(self, a: str, *args: object, **kwargs: object) -> None:
        ...

```

**VS Code extension or command-line**
Either, 1.1.103

**Additional context**
Approximation in TS: https://www.typescriptlang.org/play?ssl=20&ssc=1&pln=1&pc=1#code/MYGwhgzhAEBiD29oG8BQ0PQEZgE4AowAuaCAF1wEsA7AcwBpsTqBXAWywFNdHgStEITmGoBKEgDd4lACYp0mAL6plqUJBgAhMAC9onAB5lO1GTARI0mbHnwA6BxLAgWnCCREBPANoBdcdBSsvLWyqrqUNAAiiwG+kYmZnCIIZg4BMSkFDQM0A52Ti5uJPisHNzQAD7YgsJifgFBclZKKqhq4JExsfHGpuYpLRjphCTkVHSM+YWu7tBlXLgNktLNChhhqEA

**Comment by erictraut** [comment]
Yeah, but why would you do this? If you know the base class has a specified set of parameters with types, why not reflect them in the subclass signatures?

**Comment by layday** [comment]
I suppose because you are not interested in the values of the parameters and you are simply going to pass them through to the parent method or discard them, or if you do not wish to be bound by the implementation of the parent method.  This is quite common when overriding the constructor:

```py
class Foo:
    def __init__(self, a: str, b: int) -> None:
        ...

class Bar(Foo):
    def __init__(self, *args: object, **kwargs: object) -> None:
        super().__init__(*args, **kwargs)
        self.__some_private_attr = 1

```

**Comment by erictraut** [comment]
I guess this is technically correct and doesn't violate the Liskov substitution principle, but I wouldn't recommend doing it. 

I've updated the code to accommodate it.

**Comment by erictraut** [comment]
This is addressed in Pyright 1.1.104, which I just published. It will also be included in the next release of pylance.

**Comment by layday** [comment]
Thanks for addressing this.  I think I might've hit a bit of an edge case, which is that an error is produced if the parameters in the corresponding position on the parent class begin with an underscore:

```py
class Foo:
    def bar(self, _a: str, b: int, c: float, d: bool) -> None:
        ...

class Baz(Foo):
    # Method "bar" overrides class "Foo" in an incompatible manner
    #   Parameter 2 type mismatch: base parameter is type "str", override parameter is type "object"
    def bar(self, *args: object, **kwargs: object) -> None:
        ...
```

Conversely, this _should_ produce an error but it doesn't:

```py
class Foo:
    def bar(self, _a: str, b: int, c: float, d: bool) -> None:
        ...

class Qux(Foo):
    def bar(self, a: str, *args: object, **kwargs: object) -> None:
        ...
```

**Comment by layday** [comment]
@erictraut, in case you missed this.
